src/features/dashboard/components/AppsTable/index.tsx

```diff
@@ -266,15 +303,49 @@ const AppsTable = ({ apps }: AppsTableProps) => {
     setSelectedOptions({ ...selectedOptions, sortBy: `${columnId}${sortType}` });
   };
 
-  const getAppsTableOptionDialogProps = () => {
+  const onFilterChange = (name: string) => {
+    const { filterBy } = selectedOptions;
+    const isAllSelected = filterBy.includes('all');
+
+    if (!isAllSelected && name === 'all') {
+      setSelectedOptions((prev) => ({ ...prev, filterBy: Object.keys(tableFilterOptions) }));
+      return;
+    }
+
+    let filters = [...filterBy];
+
+    if (filters.includes(name)) {
+      filters = filters.filter((option) => option !== name && option !== 'all');
+    } else {
+      filters = [...filters, name];
+    }
+
+    const allOptionSelected = filters.length === Object.keys(tableFilterOptions).length - 1;
+    if (allOptionSelected && !filterBy.includes('all')) {
+      filters = [...filters, 'all'];
+    }
+
+    setSelectedOptions((prev) => ({ ...prev, filterBy: filters }));
+  };
+
+  const getAppsTableOptionDialogProps = useMemo(() => {
     return {
       optionType,
       isDialogOpen: isOptionDialogOpen,
       selectedOptions,
       updateSelectedOptions: setSelectedOptions,
       toggleAppTableDialog: setIsOptionDialogOpen,
     };
-  };
+  }, [isOptionDialogOpen, optionType, selectedOptions]);
+
+  const getAppsTableOptionProps = useMemo(() => {
```

Comment:
```suggestion
  const apps_table_option_props = useMemo(() => {
```

Same here @ahmed-deriv